src/components/menu/main-menu.tsx

```diff
@@ -4,81 +4,152 @@ import Link from "@components/elements/link";
 import SiteSearchForm from "@components/search/site-search-form";
 import useActiveTrail from "@lib/hooks/useActiveTrail";
 import useOutsideClick from "@lib/hooks/useOutsideClick";
-import {ChevronDownIcon} from "@heroicons/react/20/solid";
-import {MenuItem as MenuItemType} from "@lib/gql/__generated__/drupal.d";
-import {clsx} from "clsx";
-import {useBoolean, useEventListener} from "usehooks-ts";
-import {useCallback, useEffect, useId, useLayoutEffect, useRef, useState} from "react";
-import {usePathname} from "next/navigation";
+import { ArrowRightIcon, ChevronDownIcon, MagnifyingGlassIcon } from "@heroicons/react/20/solid";
+import { MenuItem as MenuItemType, StanfordBasicSiteSetting } from "@lib/gql/__generated__/drupal.d";
+import { clsx } from "clsx";
+import { useBoolean, useEventListener } from "usehooks-ts";
+import {
+  useCallback,
+  useEffect,
+  useId,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from "react";
+import { usePathname } from "next/navigation";
+import Button from "@components/elements/button";
 
 const menuLevelsToShow = 2;
 
-type Props = {
+type Props =
+  Omit<StanfordBasicSiteSetting, "__typename" | "id" | "metatag"> & {
```

Comment:
Just define and pass the 2 fields instead of the whole thing. Especially since this is a `"use client"` component, we want to only pass the most minimal data needed to work.